docs/migration/cosmoshub-4-delta-upgrade.md

```diff
@@ -0,0 +1,124 @@
+- [Delta upgrade steps](#delta-upgrade-steps)
+- [Upgrade duration](#upgrade-duration)
+- [Rollback plan](#rollback-plan)
+- [Communications](#communications)
+- [Risks](#risks)
+- [FAQ](#faq)
+
+## On-chain governance proposal attains consensus
+
+[Proposal #51](https://www.mintscan.io/cosmos/proposals/51) is the reference on-chain governance proposal for the this upgrade, which has passed with overwhleming community support. Neither core developers nor core funding entities control the governance, and this governance proposal has passed in a _fully decentralized_ way.  
+
+## Upgrade will take place July 12, 2021
+
+The upgrade will take place at a block height of `6910000`. At current block times (around 7s/block), this block height corresponds approximately to `Mon Jul 12 2021 11:00:00 GMT+0000`. This date/time is approximate as blocks are not generated at a constant interval.
+
+## Chain-id will remain the same
+
+The chain-id of the network will remain the same, `cosmoshub-4`. This is because an in-place migration of state will take place, i.e., this upgrade does not export any state.
+
+## Preparing for the upgrade
+
+### Backups
+
+Prior to the upgrade, validators are encouraged to take a full data snapshot. Snapshotting depends heavily on infrastructure, but generally this can be done by backing up the `.gaia` directory.
+
+It is critically important for validator operators to back-up the `.gaia/data/priv_validator_state.json` file after stopping the gaiad process. This file is updated every block as your validator participates in consensus rounds. It is a critical file needed to prevent double-signing, in case the upgrade fails and the previous chain needs to be restarted.
+
+### Testing
+
+For those validator and full node operators that are interested in ensuring preparedness for the impending upgrade, complete and detailed testing instructions are provided in the [gravity-dex-upgrade-test](https://github.com/b-harvest/gravity-dex-upgrade-test/) Github repository. This repository has been tested by members of the core Cosmos ecosystem, as well as ecosystem partners which include validators, exchanges, and service providers.
+
+### Public testnet
+
+Validator and full node operators that wish to test their systems on a public testnet are encouraged to join the Tendermint team's public testnet, described [here](https://github.com/b-harvest/gravity-dex-upgrade-test/#public-testnet-info).
+
+### Current runtime, cosmoshub-4 (pre-Delta upgrade) is running Gaia v4.2.1
+
+The Cosmos Hub mainnet network, `cosmoshub-4`, is currently running [Gaia v4.2.1](https://github.com/cosmos/gaia/releases/tag/v4.2.1). We anticipate that operators who are running earlier versions of Gaia, e.g., v4.2.x, will be able to upgrade successfully; however, this is untested and it is up to operators to ensure that their systems are capable of performing the upgrade. 
```

Comment:
if the current gaia is `v4.2.1` then isn't it correct that the only other version they might be running is `v4.2.0`? Or is this meant to include the RCs as well?

Reply:
Gaia has v4.1.x and v4.0.x versions available, and these are not breaking per se.